iosxr_store_box.py

```diff
@@ -108,6 +108,8 @@ def main(argv):
                         help='Optionally specify a reason for uploading this box')
     parser.add_argument('-r', '--release', action='store_true',
                         help="upload to '$ARTIFACTORY_LOCATION_RELEASE' rather than '$ARTIFACTORY_LOCATION_SNAPSHOT'.")
+    parser.add_argument('-s', '--sub_dir', dest='subdir', required=True,
```

Comment:
Given that argparse is smart enough to allow parameter abbreviations, consider `--subdirectory` instead of `--sub_dir`:

```python
>>> import argparse
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument('-s', '--subdirectory', dest='subdir')
_StoreAction(option_strings=['-s', '--subdirectory'], dest='subdir', nargs=None, const=None, default=None, type=None, choices=None, help=None, metavar=None)
>>> parser.parse_args(['--sub', 'foo'])
Namespace(subdir='foo')
>>> parser.parse_args(['--subdir', 'bar'])
Namespace(subdir='bar')
>>> parser.parse_args(['--subdirectory', 'baz'])
Namespace(subdir='baz')
```